### PR TITLE
Release 0.18.0

### PR DESCRIPTION
**Pull request by jtcohen6**
## Description & motivation
* Release v0.18.0 changes from `next` into `current`
* Includes changes from: #302, #314, #315, #319, #323, #329, #339, #328, #344, #286, #363
* Update v0.18.0 migration guide

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -4,4 +4,5 @@
   
 [context.next.environment]
   PRERELEASE = "true"
-  ALGOLIA_INDEX_NAME = "dbt-next"
+  ALGOLIA_API_KEY = "044414e92a8b8137a33fd276d4533895"
+  ALGOLIA_INDEX_NAME = "dbt_next"
```
Comment (clrcrl) [inline netlify.toml:8]: Do we need to remove this?

**Review by clrcrl — APPROVED** [review]
I noticed that the header bar still has the pre-release banner. Other than that, LGTM!

File: website/docs/docs/guides/migration-guide/upgrading-to-0-18-0.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+**Docs**
+- [project-level overviews](documentation#custom-project-level-overviews)
+
+**Redshift**
+- [`iam_profile`](redshift-profile#specifying-an-iam-profile)
+
+**Snowflake**
+- `query_tag` in [profile](snowflake-profile), [model config](snowflake-configs#query-tags)
+
+**BigQuery** - - [`impersonate_service_account`](https://docs.getdbt.com/reference/warehouse-profiles/bigquery-profile#service-account-impersonation)
```
Comment (drewbanin) [inline website/docs/docs/guides/migration-guide/upgrading-to-0-18-0.md]:
```suggestion
**BigQuery**
- [`impersonate_service_account`](https://docs.getdbt.com/reference/warehouse-profiles/bigquery-profile#service-account-impersonation)
```

**Review by drewbanin — COMMENTED** [review]
One comment and also a request to remove the "prerelease" banner for prod deployment

**Comment by jtcohen6** [comment]
@clrcrl @drewbanin Thanks for the review! No need to make any changes re: the Algolia values or the prerelease banner. Those are conditional in Netlify based on whether it is building the `next` branch, so they won't affect docs.getdbt.com.

**Pull request merged**